applicationconfig/client/README.md

```diff
@@ -0,0 +1,95 @@
+- Apache Maven 3 or above
+- Azure Service Principal
+- App Configuration Store
+
+### Adding the package to your product
+
+```xml
+<dependency>
+    <groupId>com.azure</groupId>
+    <artifactId>azure-app-configuration</artifactId>
+    <version>1.0.0-SNAPSHOT</version>
+</dependency>
+```
+
+### Authenticate the client
+
+```java
+String connectionString = "endpoint={endpoint_value};id={id_value};secret={secret_value}";
+ConfigurationClient client = ConfigurationClient.builder()
+        .credentials(new ConfigurationClientCredentials(connectionString)
+        .build();
+```
+
+Using the Azure web portal to open your App Configuration Store's Access Key section, choose one of the two connection strings and use it to replace the string value above.
+
+## Key concepts
+
+### [Key-value store](https://docs.microsoft.com/en-us/azure/azure-app-configuration/concept-key-value)
+
+Configurations are stored using key-value pairs. Keys must be unique for each configuration so it's common practice to organize keys into hierarchical namespaces using delimiting characters, for example AppName:Service1:DB:Endpoint. Optionally a label attribute may be given to a key to differentiate two configurations, for example having different versions of the configuration. Values are stored as unicode strings.
+
+### [Point-in-time snapshot](https://docs.microsoft.com/en-us/azure/azure-app-configuration/concept-point-time-snapshot)
+
+When a configuration is modified a point-in-time snapshot is recorded, snapshots are retained for seven days.
+
+### Responses
+
+```Java
+public class ConfigurationSetting {
+    private String key;
+    private String label;
+    private String value;
+    private String contentType;
+    private String etag;
+    private OffsetDateTime lastModified;
+    private boolean locked;
+    private Map<String, String> tags;
+}
+```
+
+Successful requests to the service will have a response containing a ConfigurationSetting as the body. ConfigurationSetting contains the key-value and metadata about the configuration.
+
+An unsuccessful request will throw an error that indicates why the request failed. (todo - update this with further error type explanation when implemented)
+
+For further information about responses view the [Rest API reference](https://github.com/Azure/AppConfiguration#rest-api-reference).
+
+## Examples
+
+### Create a new setting configuration
+
+### Update an existing setting configuration
+
+### Retrieve a setting configuration
+
+### Delete a setting configuration
+
+## Troubleshooting
+
+### *TODO - Add Exception Response Type*
+
+Making requests with an unauthenticated client results in the service returning an unauthorized response (401), verify that the credentials used in the construction of the client are correct.
+
+### *TODO - Add Exception Response Type*
+
+Attempting to update or delete a locked configuration results in a conflict response (409), unlock the configuration to resolve this issue.
```

Comment:
Lock and unlock functionality has been deleted for the API. See PR https://github.com/Azure/azure-sdk-for-java/pull/3319

Reply:
I feel indifferent about removing this, while the functionality was removed from the API users could still run into locked setting configurations which would result in the error response.

If we believe that it is uncommonly for setting configurations to be locked I will go ahead and remove mention of them from this section.

Reply:
how will they lock a setting?

Reply:
Removing this functionality from the API forces customers to lock/unlock only by REST calls. Notice that the Azure Portal doesn't support lock/unlock either.

It is an uncommon scenario

Reply:
The portal does allow for locking/unlocking but I agree it should be relatively uncommon, will remove the section.

![image](https://user-images.githubusercontent.com/48699787/55837405-26162f00-5ad6-11e9-89a6-cd0a608a493c.png)

Reply:
Ohh they added it. Cool :)